### PR TITLE
Xunit runner now runs multiple assemblies.

### PR DESCRIPTION
**Pull request by patriksvensson**

### CODE REVIEW

File: src/Cake.Common/Tools/XUnit/XUnit2Runner.cs [hunk 1]
```diff
@@ -55,18 +56,19 @@ public void Run(FilePath assemblyPath, XUnit2Settings settings)
                 }
             }
 
-            Run(settings, GetArguments(assemblyPath, settings));
+            var assemblies = assemblyPaths as FilePath[] ?? assemblyPaths.ToArray();
+            Run(settings, GetArguments(assemblies, settings));
         }
 
-        private ProcessArgumentBuilder GetArguments(FilePath assemblyPath, XUnit2Settings settings)
+        private ProcessArgumentBuilder GetArguments(IReadOnlyList<FilePath> assemblies, XUnit2Settings settings)
```
Comment (gep13) [inline src/Cake.Common/Tools/XUnit/XUnit2Runner.cs]: Should `assembles` not be `assemblyPaths`?

File: src/Cake.Common/Tools/XUnit/XUnitRunner.cs [hunk 2]
```diff
@@ -54,18 +56,19 @@ public void Run(FilePath assemblyPath, XUnitSettings settings)
                 }
             }
 
-            Run(settings, GetArguments(assemblyPath, settings));
+            var assemblies = assemblyPaths as FilePath[] ?? assemblyPaths.ToArray();
+            Run(settings, GetArguments(assemblies, settings));
         }
 
-        private ProcessArgumentBuilder GetArguments(FilePath assemblyPath, XUnitSettings settings)
+        private ProcessArgumentBuilder GetArguments(IReadOnlyList<FilePath> assemblies, XUnitSettings settings)
```
Comment (gep13) [inline src/Cake.Common/Tools/XUnit/XUnitRunner.cs]: Same as above, should `assemblies` not be `assemblyPaths`?

File: src/Cake.Common/Tools/XUnit/XUnitRunner.cs [hunk 3]
```diff
@@ -119,5 +122,12 @@ protected override System.Collections.Generic.IEnumerable<string> GetToolExecuta
         {
             return new[] { "xunit.console.clr4.exe" };
         }
+
+        private static FilePath GetReportName(IReadOnlyList<FilePath> assemblies)
```
Comment (gep13) [inline src/Cake.Common/Tools/XUnit/XUnitRunner.cs]: Can we avoid duplication of this method in both the `XUnitRunner` and `XUnit2Runner`?

**Comment by gep13** [comment]
@patriksvensson just had a look over this.  Apart from a few comments, this looks good to me.  Let me know when you are ready, and want to remove the WIP, and I can get it merged in.

**Comment by patriksvensson** [comment]
@gep13 Cool :smile: Not at a computer at the moment but will take a look at it during my lunch break tomorrow.

**Comment by gep13** [comment]
LGTM! :+1:

**Comment by gep13** [comment]
This relates to #688

**Pull request merged**